File: frontend/erpclink-web/src/app/features/medications/models/medication.models.ts
export interface MedicationDto {
  id: string;
  organizationId: string;
  code: string;
  name: string;
  genericName: string | null;
  strength: string | null;
  dosageForm: string | null;
  route: string | null;
  isActive: boolean;
  createdAtUtc: string;
  createdBy: string | null;
  updatedAtUtc: string | null;
  updatedBy: string | null;
}

export interface PagedMedicationsResult {
  items: MedicationDto[];
  page: number;
  pageSize: number;
  totalCount: number;
}

export interface CreateMedicationRequest {
  code: string;
  name: string;
  genericName?: string | null;
  strength?: string | null;
  dosageForm?: string | null;
  route?: string | null;
}

export interface UpdateMedicationRequest {
  name: string;
  genericName?: string | null;
  strength?: string | null;
  dosageForm?: string | null;
  route?: string | null;
}

export interface SearchMedicationsParams {
  q?: string;
  code?: string;
  name?: string;
  genericName?: string;
  strength?: string;
  dosageForm?: string;
  activeOnly?: boolean;
  page?: number;
  pageSize?: number;
}

File: src/Modules/Prescriptions/ErpClink.Modules.Prescriptions.Application/Medications/Models/MedicationModels.cs
namespace ErpClink.Modules.Prescriptions.Application.Medications.Models;

public sealed record CreateMedicationRequest(
    string Code,
    string Name,
    string? GenericName,
    string? Strength,
    string? DosageForm,
    string? Route);

public sealed record UpdateMedicationRequest(
    string Name,
    string? GenericName,
    string? Strength,
    string? DosageForm,
    string? Route);

public sealed record SearchMedicationsRequest(
    string? Q,
    string? Code,
    string? Name,
    string? GenericName,
    string? Strength,
    string? DosageForm,
    bool? ActiveOnly,
    int Page = 1,
    int PageSize = 20);

public sealed record MedicationDto(
    Guid Id,
    Guid OrganizationId,
    string Code,
    string Name,
    string? GenericName,
    string? Strength,
    string? DosageForm,
    string? Route,
    bool IsActive,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy);

public sealed record PagedMedicationsResult(
    IReadOnlyList<MedicationDto> Items,
    int Page,
    int PageSize,
    int TotalCount);
